ironic-lib/0002-Base-classes-and-examples.patch

```diff
@@ -0,0 +1,2283 @@
++# you may not use this file except in compliance with the License.
++# You may obtain a copy of the License at
++#
++#    http://www.apache.org/licenses/LICENSE-2.0
++#
++# Unless required by applicable law or agreed to in writing, software
++# distributed under the License is distributed on an "AS IS" BASIS,
++# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
++# implied.
++# See the License for the specific language governing permissions and
++# limitations under the License.
++
++from ironic_lib.system_installer.base import Setup
++
++
++class FilesystemSetup(Setup):
++    """Basic disk formatting implementation"""
++
++    conf_key = 'filesystems'
++
++    def get_src_names(self):
++        return list(self.conf)
+diff --git a/ironic_lib/system_installer/lvmsetup.py b/ironic_lib/system_installer/lvmsetup.py
+new file mode 100644
+index 0000000..890c95d
+--- /dev/null
++++ b/ironic_lib/system_installer/lvmsetup.py
+@@ -0,0 +1,28 @@
++# Copyright (c) 2018 Intel Corporation
++#
++# Licensed under the Apache License, Version 2.0 (the "License");
++# you may not use this file except in compliance with the License.
++# You may obtain a copy of the License at
++#
++#    http://www.apache.org/licenses/LICENSE-2.0
++#
++# Unless required by applicable law or agreed to in writing, software
++# distributed under the License is distributed on an "AS IS" BASIS,
++# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
++# implied.
++# See the License for the specific language governing permissions and
++# limitations under the License.
++
++from ironic_lib.system_installer.base import Setup
++
++
++class LvmSetup(Setup):
++    """LVM setup implementation"""
++
++    conf_key = 'lvm'
++
++    def get_src_names(self):
++        return sum([list(v['PVs']) for v in self.conf.values()], [])
++
++    def get_dst_names(self):
++        return sum([list(v['LVs']) for v in self.conf.values()], [])
+diff --git a/ironic_lib/system_installer/megaclisetup.py b/ironic_lib/system_installer/megaclisetup.py
+new file mode 100644
+index 0000000..b32d878
+--- /dev/null
++++ b/ironic_lib/system_installer/megaclisetup.py
+@@ -0,0 +1,22 @@
++# Copyright (c) 2018 Intel Corporation
++#
++# Licensed under the Apache License, Version 2.0 (the "License");
++# you may not use this file except in compliance with the License.
++# You may obtain a copy of the License at
++#
++#    http://www.apache.org/licenses/LICENSE-2.0
++#
++# Unless required by applicable law or agreed to in writing, software
++# distributed under the License is distributed on an "AS IS" BASIS,
++# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
++# implied.
++# See the License for the specific language governing permissions and
++# limitations under the License.
++
++from ironic_lib.system_installer.base import Setup
++
++
++class MegaCliSetup(Setup):
++    """Hardware RAID setup implementation"""
++
++    conf_key = 'megacli'
```

Comment:
I assume the rest is implemented in a future patch?

Reply:
Yes, in https://github.com/yahoo/openstack/pull/13, generally we implemented each of those classes/functionalities in separate PRs

Reply:
:+1: